drivers/flash/flash_spi.h

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright (C) 2021 Westberry Technology (ChangZhou) Corp., Ltd
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+/* All the following default configurations are based on MX25L4006E Nor FLASH. */
+
+/*
+    The slave select pin of the FLASH.
+    This needs to be a normal GPIO pin_t value, such as B14.
+*/
+#ifndef EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN
+#    error "No chip select pin defined -- missing EXTERNAL_FLASH_SPI_SLAVE_SELECT_PIN"
+#endif
+
+/*
+    The clock divisor for SPI to ensure that the MCU is within the
+    specifications of the FLASH chip. Generally this will be PCLK divided by
+    the intended divisor -- check your clock settings and the datasheet of
+    your FLASH.
+*/
+#ifndef EXTERNAL_FLASH_SPI_CLOCK_DIVISOR
+#    ifdef __AVR__
+#        define EXTERNAL_FLASH_SPI_CLOCK_DIVISOR 8
+#    else
+#        define EXTERNAL_FLASH_SPI_CLOCK_DIVISOR 18
+#    endif
+#endif
```

Comment:
```suggestion
#ifndef EXTERNAL_FLASH_SPI_CLOCK_DIVISOR
#    ifdef __AVR__
#        define EXTERNAL_FLASH_SPI_CLOCK_DIVISOR 4
#    else
#        define EXTERNAL_FLASH_SPI_CLOCK_DIVISOR 8
#    endif
#endif
```
This should be a power of 2. The reference MCU used is ATMEGA32U4 for AVRs, and STM32F303 for ARMs. The above values will result in 4MHz clock for 32U4, and 4.5MHz clock for F303 with the default board definitions.

Keyboards are then free to override as they see fit.